include/sempr/entity/spatial/AbstractPointCloud.hpp

```diff
@@ -119,18 +120,19 @@ class AbstractPointCloud
      * typename std::enable_if<std::is_fundamental<D>::value>
      * virtual AbstractChannel<D>& getChannel(int type) = 0;
      */
-    virtual void getChannel(int type, AbstractChannel<int8_t>& channel) const = 0;
-    virtual void getChannel(int type, AbstractChannel<int16_t>& channel) const = 0;
-    virtual void getChannel(int type, AbstractChannel<int32_t>& channel) const = 0;
-    virtual void getChannel(int type, AbstractChannel<int64_t>& channel) const = 0;
-
-    virtual void getChannel(int type, AbstractChannel<uint8_t>& channel) const = 0;
-    virtual void getChannel(int type, AbstractChannel<uint16_t>& channel) const = 0;
-    virtual void getChannel(int type, AbstractChannel<uint32_t>& channel) const = 0;
-    virtual void getChannel(int type, AbstractChannel<uint64_t>& channel) const = 0;
-
-    virtual void getChannel(int type, AbstractChannel<float>& channel) const = 0;
-    virtual void getChannel(int type, AbstractChannel<double>& channel) const = 0;
+    virtual void getChannel(int type, AbstractChannel<int8_t>&& channel) = 0;
+    virtual void getChannel(int type, AbstractChannel<int16_t>&& channel) = 0;
+    virtual void getChannel(int type, AbstractChannel<int32_t>&& channel) = 0;
+    virtual void getChannel(int type, AbstractChannel<int64_t>&& channel) = 0;
```

Comment:
Yes its is. This shall use the default move constructor of the based vector.
This will avoid a copy of the whole channel.

Its a bit tricky and some part of modern C++ (11+) but it is less ugly than `**` ;-)

Reply:
But a move-constructor will leave the source in an undefined state! So when using `cloud->getChannel(...)` afterwards **I** got the channel, but the pointcloud lost it?!

Please clarify, what is the supposed semantics of `getChannel`? I thought it should return a reference / pointer to the channel of the pointcloud, so if I modify the channel, I modify the channel in the cloud. But the ownership stays at the cloud. Correct?